### PR TITLE
MultiStepTDLoss

### PR DESCRIPTION
**Pull request by emailweixu**

### CODE REVIEW

File: alf/algorithms/multi_step_td_loss.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+        Sutton et al. `Reinforcement Learning: An Introduction
+        <http://incompleteideas.net/book/the-book.html>`_, Chapter 12, 2018
+
+        Args:
+            td_errors_loss_fn (Callable): A function for computing the TD errors
+                loss. This function takes as input the target and the estimated
+                Q values and returns the loss for each element of the batch.
+            td_lambda (float): Lambda parameter for TD-lambda computation.
+            td_loss_weight (float): the weigt for the loss of td error.
```
Comment (Haichao-Zhang) [inline alf/algorithms/multi_step_td_loss.py]: Doc needs a little update. ```gamma``` needs to be added and ```td_loss_weight``` can be removed.

File: alf/algorithms/multi_step_td_loss.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+import alf
+from alf.data_structures import TrainingInfo, LossInfo, StepType
+from alf.utils.losses import element_wise_squared_loss
+from alf.utils import tensor_utils, value_ops
+from alf.utils.summary_utils import safe_mean_hist_summary
+
+
+@gin.configurable
+class MultiStepTDLoss(nn.Module):
```
Comment (hnyu) [inline alf/algorithms/multi_step_td_loss.py:27]: It seems that this loss has quite some overlap with ActorCriticLoss?

File: alf/algorithms/multi_step_td_loss.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                 gamma=0.99,
+                 td_error_loss_fn=element_wise_squared_loss,
+                 td_lambda=0.95,
+                 debug_summaries=False,
+                 name="MultiStepTDLoss"):
+        r"""Create a ActorCriticLoss object.
+
+        Let :math:`G_{t:T}` be the bootstaped return from t to T:
+            :math:`G_{t:T} = \sum_{i=t+1}^T \gamma^{t-i-1}R_t + \gamma^{T-t} V(s_T)`
+        If ``td_lambda`` = 1, the target for step t is :math:`R_{t:T}`.
```
Comment (hnyu) [inline alf/algorithms/multi_step_td_loss.py]: R_{t:T} -> G_{t:T}

File: alf/algorithms/multi_step_td_loss.py (hunk 2 above)
Comment (emailweixu) [reply alf/algorithms/multi_step_td_loss.py:27]: I am aware of it. It's also similar to OneStepTDLoss. I choose not to extract common part to make things clean.

**Review by Haichao-Zhang — APPROVED** [review]
